src/main/js/components/header/actions-overflow.js

```diff
@@ -0,0 +1,127 @@
+import Utils from "@/components/dropdowns/utils";
+import { createElementFromHtml } from "@/util/dom";
+
+const OVERFLOW_ID = "jenkins-header-actions-overflow";
+
+export default function computeActions() {
+  document
+    .querySelectorAll(
+      ".jenkins-header__actions .jenkins-button[data-type='header-action'].jenkins-hidden",
+    )
+    .forEach((e) => {
+      e.classList.remove("jenkins-hidden");
+    });
+
+  if (!actionsOverflows()) {
+    removeOverflowButton();
+    return;
+  }
+
+  const items = [];
+  const actions = Array.from(
+    document.querySelectorAll(
+      ".jenkins-header__actions .jenkins-button[data-type='header-action']",
+    ),
+  ).slice(1, -1);
+
+  const overflowButton = generateOverflowButton();
+
+  while (actionsOverflows()) {
+    const item = actions.pop();
+
+    if (!item) {
+      break;
+    }
+
+    items.unshift(item);
+    item.classList.add("jenkins-hidden");
+  }
+
+  Utils.generateDropdown(
+    overflowButton,
+    (instance) => {
+      const mappedItems = items.map((e) => {
+        let icon = e.querySelector("img");
+        if (icon) {
+          icon = icon.src;
+        }
+        let iconXml = e.querySelector("svg");
+        if (iconXml) {
+          icon = true;
+          iconXml = iconXml.outerHTML;
+        }
+
```

Comment:
```suggestion

        const span = e.querySelector(".app-action-name");
        let label = e.textContent;
        if (span !== null) {
          label = span.textContent;
        }

```
Avoids that any textContent in the SVG is included

Reply:
Thanks! Changed it to use `data-type` rather than the class - but working identically.